control/controlclient/direct.go

```diff
@@ -1303,3 +1303,59 @@ func (c *Direct) SetDNS(ctx context.Context, req *tailcfg.SetDNSRequest) error {
 
 	return nil
 }
+
+// tsmpPing sends a Ping to pr.IP, and sends an http request back to pr.URL
+// with ping response data.
+func tsmpPing(logf logger.Logf, c *http.Client, pr *tailcfg.PingRequest, pinger Pinger) error {
+	var err error
+	if pr.URL == "" {
+		return errors.New("invalid PingRequest with no URL")
```

Comment:
In general, it's nice to put the various errors in exported package variables so that the caller can test for what the error is rather than just printing it out. In this case, it doesn't really matter, we can adjust that later if needed.